src/test/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GermlineCNVSegmentVariantComposerUnitTest.java

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.hellbender.tools.copynumber.gcnv;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import org.apache.commons.math3.util.FastMath;
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.IntegerCopyNumberSegmentCollection;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.IntegerCopyNumberSegmentCollectionUnitTest;
+import org.broadinstitute.hellbender.tools.copynumber.formats.records.IntegerCopyNumberSegment;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Unit test for {@link GermlineCNVSegmentVariantComposer}.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public class GermlineCNVSegmentVariantComposerUnitTest extends GATKBaseTest {
+    @Test(dataProvider = "variantCompositionSettings")
+    public void testVariantComposition(final int refAutosomalCopyNumber,
```

Comment:
I would also add a more universal test that tests that creates an actual segment VCF and checks its correctness

Reply:
This is a test for the correctness of the generated `VariantContext`. Testing an arbitrary `VariantContext` is written correctly to VCF is the responsibility of the author of `VariantContextWriter`.